src/models/tabularmodel.jl

```diff
@@ -0,0 +1,53 @@
+function emb_sz_rule(n_cat)
+	min(600, round(1.6 * n_cat^0.56))
+end
+
+function _one_emb_sz(catdict, catcol::Symbol, sz_dict=nothing)
+	sz_dict = isnothing(sz_dict) ? Dict() : sz_dict
+	n_cat = length(catdict[catcol])
+	sz = catcol in keys(sz_dict) ? sz_dict[catcol] : emb_sz_rule(n_cat)
+	Int64(n_cat), Int64(sz)
+end
+
+function get_emb_sz(catdict, cols; sz_dict=nothing)
+	[_one_emb_sz(catdict, catcol, sz_dict) for catcol in cols]
+end
+
+# function get_emb_sz(td::TableDataset, sz_dict=nothing)
+# 	cols = Tables.columnaccess(td.table) ? Tables.columnnames(td.table) : Tables.columnnames(Tables.rows(td.table)[1])
+# 	[_one_emb_sz(catdict, catcol, sz_dict) for catcol in cols]
+# end
+
+function TabularModel(
+		layers; 
+		emb_szs,
+		n_cont::Int64,
+		out_sz,
+		ps::Union{Tuple, Vector, Number}=0,
+		embed_p::Float64=0.,
+		y_range=nothing,
+		use_bn::Bool=true,
+		bn_final::Bool=false,
+		bn_cont::Bool=true,
+		act_cls=Flux.relu,
+		lin_first::Bool=true)
+
+	embedslist = [Embedding(ni, nf) for (ni, nf) in emb_szs]
+	emb_drop = Dropout(embed_p)
+	embeds = Chain(x -> ntuple(i -> x[i, :], length(emb_szs)), Parallel(vcat, embedslist...), emb_drop)
+
+	bn_cont = bn_cont ? BatchNorm(n_cont) : identity
+
+	n_emb = sum(size(embedlayer.weight)[1] for embedlayer in embedslist)
+	sizes = append!(zeros(0), [n_emb+n_cont], layers)
+	actns = append!([], [act_cls for i in 1:(length(sizes)-1)])
+
+	_layers = []
+	for (i, (p, a)) in enumerate(zip(Iterators.cycle(ps), actns))
+		layer = linbndrop(Int64(sizes[i]), Int64(sizes[i+1]), use_bn=use_bn, p=p, act=a, lin_first=lin_first)
+		push!(_layers, layer)
+	end
+	push!(_layers, linbndrop(Int64(last(sizes)), Int64(out_sz), use_bn=bn_final, lin_first=lin_first))
```

Comment:
```suggestion
        n_cat = first(Flux.outputsize(embeds, (length(emb_szs), 1)))
	classifiers = [
	    linbndrop(n_cat, first(layers); use_bn=use_bn, p=first(ps), act=act_cls, lin_first=lin_first),
	    [linbndrop(isize, osize; use_bn=use_bn, p=p, act=act_cls, lin_first=lin_first)
	      for (isize, osize, p) in zip(layers[1:(end - 2)], layers[2:(end - 1)], Iterators.cycle(Base.tail(ps)))]...,
	    linbndrop(last(layers)), out_sz; use_bn=bn_final, lin_first=lin_first)
	]
```

Reply:
I would go so far as to convert the comprehension into a plain for loop with `push!`. Not any less efficient than the splat and would help with line lengths.

Reply:
Yeah maybe that's better here since we are using so many iterables in the comprehension

Reply:
Seeing that `Flux.outputsize` isn't working with `Embedding` layers, I was thinking of keeping the current way of calculating `n_emb` for now.